File: core/src/com/alta_v2/mediator/serde/UpdaterFactory.java
package com.alta_v2.mediator.serde;

import com.alta_v2.mediator.screen.menu.MenuScreenUpdater;
import com.alta_v2.mediator.screen.tiledMap.TiledMapScreenUpdater;
import com.alta_v2.physics.TiledMapPhysicEngine;

/**
 * Provides the factory for creating of screen updaters.
 */
public interface UpdaterFactory {

    /**
     * Creates the {@link MenuScreenUpdater} instance.
     */
    MenuScreenUpdater createMenuScreenUpdater();

    /**
     * Creates the {@link TiledMapScreenUpdater} instance.
     */
    TiledMapScreenUpdater createTiledMapScreenUpdater(TiledMapPhysicEngine engine);
}

File: core/src/com/alta_v2/mediator/serde/UpdaterFactory.kt
package com.alta_v2.mediator.serde

import com.alta_v2.mediator.screen.menu.MenuScreenUpdater
import com.alta_v2.mediator.screen.tiledMap.TiledMapScreenUpdater
import com.alta_v2.physics.TiledMapPhysicEngine

/**
 * Provides the factory for creating of screen updaters.
 */
interface UpdaterFactory {
    /**
     * Creates the [MenuScreenUpdater] instance.
     */
    fun createMenuScreenUpdater(): MenuScreenUpdater

    /**
     * Creates the [TiledMapScreenUpdater] instance.
     */
    fun createTiledMapScreenUpdater(engine: TiledMapPhysicEngine): TiledMapScreenUpdater
}